website/docs/feature-flag-tutorials/django/implementing-feature-flags-django.md

```diff
@@ -0,0 +1,183 @@
+
+```
+git clone https://github.com/unleash/unleash.git
+cd unleash
+docker compose up -d
+```
+
+You will now have Unleash installed onto your machine and running in the background. You can access this instance in your web browser at [http://localhost:4242](http://localhost:4242).
+
+Log in to the platform using these credentials:
+
+```
+Username: admin
+Password: unleash4all
+```
+
+Click the 'New feature flag' button to create a new feature flag.
+
+![Create a new feature flag](../ruby/new-ff.png)
+
+Call it `top-3` and enable it in the `development` environment.
+
+![A feature flag called `top-3` is now visible.](../rails/enable-ff.png)
+
+Everything's now set up on the Unleash side. Let's set up the Django application.
+
+## 3. Setup the Django app
+
+Let's clone a basic blog repository and get it up and running. We don't want to waste time setting up a Django codebase from scratch.
+
+```sh
+git clone https://github.com/alvinometric/django-basic-blog
+cd django-basic-blog
+```
+
+#### Setup/seed the database
+
+This repository uses SQLite, so no additional dependencies or migrations are required.
+
+```sh
+python manage.py loaddata initial_data.json
+```
+
+#### Run the server
+
+```sh
+python manage.py runserver
+```
+
+Go to [http://localhost:8000](http://localhost:8000) and check that you see the following:
+
+![A blog app with a list of posts](../rails/blog-app.png)
+
+## 4. Restrict the number of posts
+
+Right now all the blog posts are displayed on the index page. We want to use a feature flag to change that and restrict it to the 3 most recent posts.
+
+Let's create a static boolean flag, for now.
+
+Modify the `post_list` view in `blog/views.py` to look like this:
+
+```python
+from django.shortcuts import render
+from .models import Post
+
+def post_list(request):
+    is_top3 = True
+    posts = Post.objects.order_by('-created_at')[:3] if is_top3 else Post.objects.all()
+    return render(request, 'blog/post_list.html', {'posts': posts})
+```
+
+We're using the flag in the views rather than the template, but you could also do it there. I prefer keeping my template logic as simple as possible.
+
+Reload your browser and you should see only the 3 most recent posts.
+
+## 5. Add Unleash to your Django app
+
+Now, let's connect our project to Unleash so that you can toggle the feature flag at runtime. If you wanted to, you could also do a gradual rollout, and use it for A/B testing or more advanced functionality.
+
+You'll need 2 things:
+
+-   The URL of your Unleash instance's API. It's `http://localhost:4242/api/` for your local version. You'll want to replace it with your remote instance.
+-   The API token we created on our Unleash instance.
+
+First, install the `UnleashClient` package:
+
+```sh
+pip install UnleashClient
+```
+
+Then, create `blog/unleash_client.py`, and add the following:
+
+```python
+from UnleashClient import UnleashClient
```

Comment:
Starting the client in the view is definitely not sane for a production config. I think that's totally sane for a tutorial and going through a dance of setting up a singleton is probably out of scope here but we should make this clear that this does belong in a singleton and not newed up where ever